src/vs/workbench/contrib/chat/browser/chatContentParts/toolInvocationParts/chatTerminalToolConfirmationSubPart.ts

```diff
@@ -114,19 +119,26 @@ export class TerminalConfirmationWidgetSubPart extends BaseChatToolInvocationSub
 				ariaLabel: typeof title === 'string' ? title : title.value
 			}
 		};
-		const langId = this.languageService.getLanguageIdByLanguageName(terminalData.language ?? 'sh') ?? 'shellscript';
+		const languageId = this.languageService.getLanguageIdByLanguageName(terminalData.language ?? 'sh') ?? 'shellscript';
 		const model = this.modelService.createModel(
 			terminalData.commandLine.toolEdited ?? terminalData.commandLine.original,
-			this.languageService.createById(langId),
+			this.languageService.createById(languageId),
 			this._getUniqueCodeBlockUri(),
 			true
 		);
+		textModelService.createModelReference(model.uri).then(ref => {
```

Comment:
See my helper `thenifNotDisposed`

Reply:
The fact that we have to do this is confusing.

Reply:
I don't think I can use that since I need to call `dispose` on it to free the reference when the chat part is disposed?

Reply:
Good point. There could probably be a helper that covers what you want to do